src/app/lib/main.dart

```diff
@@ -153,8 +185,8 @@ void _registerRepositories() {
 void _registerServices() {
   GetIt.I.registerSingleton(
     DadJokesService(
-      GetIt.I.get<DadJokesRepository>(),
-      GetIt.I.get<FavoriteDadJokesRepository>(),
+      GetIt.I.get<DadJokesRepositoryDecorator>(),
```

Comment:
Here I think you should resolve the interfaces and not the concrete implementations.

Reply:
Doing this makes it not work anymore, plus on .net we register the decorator 
![image](https://github.com/nventive/FlutterApplicationTemplate/assets/72167772/609bc004-154a-47d3-b9ef-46be3ae94b29)